[BUG] 8.0.0 broken by including a non-existing header: `zmalloc.c: fatal error: threads.h: No such file or directory`

**Describe the bug**

```
/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_databases_valkey/valkey/work/compwrap/cc/opt/local/bin/gcc-mp-14 -pedantic -std=gnu11 -Wall -W -Wno-missing-field-initializers -Werror=deprecated-declarations -Wstrict-prototypes -O3 -flto=auto -fno-omit-frame-pointer -g -ggdb -arch ppc -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -I../deps/hiredis -I../deps/linenoise -I../deps/lua/src -I../deps/hdr_histogram -I../deps/fpconv -DUSE_JEMALLOC -I../deps/jemalloc/include -DUSE_OPENSSL=1  -DBUILD_TLS_MODULE=0 -MMD -o zmalloc.o -c zmalloc.c
zmalloc.c:94:10: fatal error: threads.h: No such file or directory
   94 | #include <threads.h>
      |          ^~~~~~~~~~~
compilation terminated.
make[1]: *** [zmalloc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_databases_valkey/valkey/work/valkey-8.0.0/src'
make: *** [all] Error 2
```

**To reproduce**

Run the build.

**Expected behavior**

It should build, as it did before.

**Additional information**

macOS 10.6, gcc 14.2.0

--- a/src/zmalloc.c
+++ b/src/zmalloc.c
@@ -88,11 +88,7 @@ void zlibc_free(void *ptr) {
 #define dallocx(ptr, flags) je_dallocx(ptr, flags)
 #endif
 
-#if __STDC_NO_THREADS__
-#define thread_local __thread
-#else
-#include <threads.h>
-#endif
+#define thread_local _Thread_local
 
 #define MAX_THREADS_NUM (IO_THREADS_MAX_NUM + 3 + 1)
 /* A thread-local storage which keep the current thread's index in the used_memory_thread array. */

